pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,325 @@
+}
+
+type policyRecommendationComponent struct {
+	cfg   *PolicyRecommendationConfiguration
+	image string
+}
+
+func PolicyRecommendation(cfg *PolicyRecommendationConfiguration) Component {
+	return &policyRecommendationComponent{
+		cfg: cfg,
+	}
+}
+
+func (pr *policyRecommendationComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := pr.cfg.Installation.Registry
+	path := pr.cfg.Installation.ImagePath
+	prefix := pr.cfg.Installation.ImagePrefix
+
+	var err error
+	pr.image, err = components.GetReference(components.ComponentPolicyRecommendation, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (pr *policyRecommendationComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeAny
+}
+
+func (pr *policyRecommendationComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		CreateNamespace(PolicyRecommendationNamespace, pr.cfg.Installation.KubernetesProvider, PSSRestricted),
+		allowTigeraPolicyForPolicyRecommendation(pr.cfg),
+	}
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.PullSecrets...)...)...)
+
+	objs = append(objs,
+		pr.serviceAccount(),
+		pr.clusterRole(),
+		pr.clusterRoleBinding(),
+		pr.deployment(),
+	)
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.ESSecrets...)...)...)
+
+	return objs, nil
+}
+
+func (pr *policyRecommendationComponent) Ready() bool {
+	return true
+}
+
+func (pr *policyRecommendationComponent) clusterRole() client.Object {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{"licensekeys", "managedclusters"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"crd.projectcalico.org"},
+			Resources: []string{"licensekeys"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{
+				"tiers",
+				"policyrecommendationscope",
+				"policyrecommendationscope/status",
+				"policyrecommendationscopes",
+				"policyrecommendationscopes/status",
+				"stagednetworkpolicies",
+				"tier.stagednetworkpolicies",
+				"stagednetworkpolicy",
+				"tier.stagednetworkpolicy",
+			},
+			Verbs: []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+		},
+	}
+
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PolicyRecommendationClusterRoleName,
+		},
+		Rules: rules,
+	}
+}
+
+func (pr *policyRecommendationComponent) clusterRoleBinding() client.Object {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
```

Comment:
I think this can be a RoleBinding within the policy recommendation namespace - doesn't need to be a CRB because it only binds to one serviceaccount in a single namespace.

Reply:
[ERROR][1] monitor.go 370: Failed to load product license from datastore. error=connection is unauthorized: licensekeys.crd.projectcalico.org "default" is forbidden: User "system:serviceaccount:tigera-policy-recommendation:tigera-policy-recommendation" cannot get resource "licensekeys" in API group "crd.projectcalico.org" at the cluster scope